streaming/rdm_atomic.c

```diff
@@ -87,26 +101,96 @@ static enum fi_op get_fi_op(char *op) {
 		return FI_MIN;
 	else if (!strcmp(op, "max"))
 		return FI_MAX;
+	else if (!strcmp(op, "sum"))
+		return FI_SUM;
+	else if (!strcmp(op, "prod"))
+		return FI_PROD;
+	else if (!strcmp(op, "lor"))
+		return FI_LOR;
+	else if (!strcmp(op, "land"))
+		return FI_LAND;
+	else if (!strcmp(op, "bor"))
+		return FI_BOR;
+	else if (!strcmp(op, "band"))
+		return FI_BAND;
+	else if (!strcmp(op, "lxor"))
+		return FI_LXOR;
+	else if (!strcmp(op, "bxor"))
+		return FI_BXOR;
 	else if (!strcmp(op, "read"))
 		return FI_ATOMIC_READ;
 	else if (!strcmp(op, "write"))
 		return FI_ATOMIC_WRITE;
 	else if (!strcmp(op, "cswap"))
 		return FI_CSWAP;
+	else if (!strcmp(op, "cswap_ne"))
+		return FI_CSWAP_NE;
+	else if (!strcmp(op, "cswap_le"))
+		return FI_CSWAP_LE;
+	else if (!strcmp(op, "cswap_lt"))
+		return FI_CSWAP_LT;
+	else if (!strcmp(op, "cswap_ge"))
+		return FI_CSWAP_GE;
+	else if (!strcmp(op, "cswap_gt"))
+		return FI_CSWAP_GT;
+	else if (!strcmp(op, "mswap"))
+		return FI_MSWAP;
 	else {
-		fprintf(stderr, "Not supported by the example\n");
+		fprintf(stderr, "Not a valid atomic operation\n");
 		return FI_ATOMIC_OP_LAST;
 	}
 }
 
+static inline size_t datatype_to_size(enum fi_datatype datatype)
```

Comment:
> > +static inline size_t datatype_to_size(enum fi_datatype datatype)
> 
> Can we export this function so that apps can use it?

Maybe - we would need to figure out what to do with the floating point sizes.